src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -23,6 +23,18 @@ import { getRawFetchHeaders } from '../../utils/getRawFetchHeaders'
 
 export type Protocol = 'http' | 'https'
 
+enum HttpClientInternalState {
```

Comment:
First, I'm adding a new `state` internal flag that represents the internal state of the `ClientRequest`. I take advantage of the numeric values of the enums in TypeScript to filter out all the states before a certain states like so:

```ts
if (this.state < HttpClientInternalState.MockLookupEnd) {
  // This is for states 0 - 3
}
```